### PR TITLE
Move external facing podresources apis to staging

### PR DESCRIPTION
**Pull request by RenaudWasTaken**
What type of PR is this?
/kind api-change
/area code-organization
/sig node

**What this PR does / why we need it:**
podresources API is external facing. Currently we are forcing folks to import them from k8s.io/kubernetes which is not optimal. We should instead make them available from the k8s.io/kubelet repo to make it easier for them.

**Which issue(s) this PR fixes:** N/A

**Special notes for your reviewer:**
Note, for people who will be broken by this change (i.e: people pulling from kubernetes master), they can either:
a) Freeze the kubernetes version they rely on, which they already should, in the go.mod they should have something along the lines of
```
require (
        ...
	k8s.io/kubernetes v0.17.4
        ....
)
```
b) Change the import path:
```
sed -i 's#k8s.io/kubernetes/pkg/kubelet/apis/podresources/v1alpha1#k8s.io/kubelet/pkg/apis/podresources/v1alpha1#g" *.go
go mod tidy
```

/cc @dashpole @derekwaynecarr 

Enhancement issue: https://github.com/kubernetes/enhancements/issues/606
Current Kep: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/compute-device-assignment.md
Updated Kep: https://github.com/kubernetes/enhancements/pull/1865

Note: Change is targeting current milestone and we expect to introduce v1 in the next release.
Also note that despite the API name, it is currently in beta (since 1.15).

**Does this PR introduce a user-facing change?:**
```release-note
external facing API podresources is now available under k8s.io/kubelet/pkg/apis/
```

### CODE REVIEW

**Comment by RenaudWasTaken** [comment]
/assign @dims 

Note that I followed the model of your PR: https://github.com/kubernetes/kubernetes/pull/83551

**Review by SergeyKanzhelev — APPROVED** [review]

File: hack/update-generated-pod-resources-dockerized.sh [hunk 1]
```diff
@@ -23,7 +23,7 @@ set -o nounset
 set -o pipefail
 
 KUBE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../" && pwd -P)"
-POD_RESOURCES_ALPHA="${KUBE_ROOT}/pkg/kubelet/apis/podresources/v1alpha1/"
```
[the remark this answers is not in the document]
Comment (RenaudWasTaken) [reply hack/update-generated-pod-resources-dockerized.sh:26]: Thanks let me update that!

[the remark this answers is not in the document]
Comment (RenaudWasTaken) [reply hack/update-generated-pod-resources-dockerized.sh:26]: Fixed!

**Comment by dashpole** [comment]
/lgtm
/approve

Thanks @RenaudWasTaken

**Comment by liggitt** [comment]
moving API types intended to be consumed outside k8s.io/kubernetes to a place where they can be consumed outside k8s.io/kubernetes makes sense 👍 

/lgtm
/approve

**Comment by neolit123** [comment]
please see this PR merge policy change.
https://groups.google.com/g/kubernetes-dev/c/YXGBa6pxLzo

the milestone should be applied later in the cycle.
for Q you can ask in #sig-release.

thanks
/milestone clear

**Comment by RenaudWasTaken** [comment]
I'll probably wait a bit that regular maintenance restarts before rebasing this PR :) !
It's not critical to stability and doesn't fix any tests.

**Comment by SergeyKanzhelev** [comment]
> I'll probably wait a bit that regular maintenance restarts before rebasing this PR :) !
> It's not critical to stability and doesn't fix any tests.

@RenaudWasTaken this time came =) It's now in 1.20 milestone

**Comment by RenaudWasTaken** [comment]
Will take care of it this week then!

File: pkg/kubelet/server/server.go [hunk 2]
```diff
@@ -60,12 +60,13 @@ import (
...
 	compbasemetrics "k8s.io/component-base/metrics"
 	"k8s.io/component-base/metrics/legacyregistry"
+	podresourcesapi "k8s.io/kubelet/pkg/apis/podresources/v1alpha1"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
 	api "k8s.io/kubernetes/pkg/apis/core"
 	"k8s.io/kubernetes/pkg/apis/core/v1/validation"
 	"k8s.io/kubernetes/pkg/features"
 	"k8s.io/kubernetes/pkg/kubelet/apis/podresources"
-	podresourcesapi "k8s.io/kubernetes/pkg/kubelet/apis/podresources/v1alpha1"
+	podresourcesapi "k8s.io/kubelet/pkg/apis/podresources/v1alpha1"
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/server/server.go:69]: build complains about adding it twice

Comment (RenaudWasTaken) [reply pkg/kubelet/server/server.go]: Thanks for pointing this out! Fixed 😛

**Pull request merged**